spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -75,44 +76,36 @@ protected SQLAdmin sqlAdminService(CredentialsProvider credentialsProvider)
 
 		return new SQLAdmin.Builder(httpTransport, jsonFactory,
 				new HttpCredentialsAdapter(credentialsProvider.getCredentials()))
-				.build();
+						.build();
 	}
 
 	@Bean
 	@ConditionalOnMissingBean(CloudSqlJdbcUrlProvider.class)
 	public CloudSqlJdbcUrlProvider cloudSqlJdbcUrlProvider(SQLAdmin sqlAdmin,
 			GcpProjectIdProvider projectIdProvider)
-			throws IOException, ClassNotFoundException {
+			throws IOException {
 		Assert.hasText(projectIdProvider.getProjectId(),
 				"A project ID must be provided.");
 
-		// TODO(joaomartins): Set the credential factory to be used by SocketFactory when
-		// https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues/41 is
-		// resolved. For now, it uses application default creds.
-
-		// Load MySQL driver to register it, so the user doesn't have to.
-		Class.forName("com.mysql.jdbc.Driver");
-
-		// Look for the region if the user didn't specify one.
-		String region = this.gcpCloudSqlProperties.getRegion();
-		if (!StringUtils.hasText(region)) {
-			region = sqlAdmin.instances().get(projectIdProvider.getProjectId(),
-					this.gcpCloudSqlProperties.getInstanceName()).execute().getRegion();
+		String appEngineVersion = System.getProperty("com.google.appengine.runtime.version");
```

Comment:
Nice, can the conditional match the string prefix too? This is needed because you can have the property, but w/ a different value, which means different thing. It has to match based on the string prefix.